plugin.py

```diff
@@ -77,3 +79,57 @@ def _cleanup_cache(cls) -> None:
         """ Clean up this plugin's cache directory. """
 
         shutil.rmtree(get_plugin_storage_dir(), ignore_errors=True)
+
+
+class LspTexlabForwardSearchCommand(LspTextCommand):
+
+    def run(self, edit):
+        session = self.session_by_name(PLUGIN_NAME)
+        if not session:
+            return
+        params = text_document_position_params(self.view, next(iter(self.view.sel())).a)
+        session.send_request(Request("textDocument/forwardSearch", params), self.on_response_async, self.on_error_async)
+
+    def on_response_async(self, response):
+        status = response["status"]
+        window = self.view.window()
+        if window is None:
+            return
+        if status == 0:
+            pass  # success
+        elif status == 1:
+            window.status_message(PLUGIN_NAME + ": Previewer exited with errors")
+        elif status == 2:
+            window.status_message(PLUGIN_NAME + ": Previewer failed to start or crashed")
+        elif status == 3:
+            window.status_message(PLUGIN_NAME + ": Previewer is not configured")
```

Comment:
There exists `self.weaksession` from which you can get the window. But possibly better is to get the views from `weaksession` using `session_views_async` and post a status on those views only.

Reply:
Actually `Session.set_window_status_async()` can be used even.

Reply:
Never mind those comments. Thought it's the AbstractSession implementation.